src/core/services/slot-refresher.ts

```diff
@@ -24,35 +31,123 @@ async function isUAP(): Promise<boolean> {
 	});
 }
 
-function refreshWhenBackInViewport(adSlot) {
+function callBidders(slotName: string, callback: () => void) {
+	const bidGroup = context.get(`slots.${slotName}.bidGroup`);
+
+	if (!bidGroup) {
+		context.set(`slots.${slotName}.bidGroup`, slotName);
+	}
+
+	communicationService.emit(eventsRepository.BIDDERS_CALL_PER_GROUP, {
+		group: slotName,
+		callback: callback,
+	});
+}
+
+function refreshWhenBackInViewport(adSlot: AdSlot) {
 	function refresh(event) {
 		if (event.slot.getSlotElementId() === adSlot.getSlotName()) {
 			logger(logGroup, `${adSlot.getSlotName()} back in the viewport, refreshing.`, event);
-			GptProvider.refreshSlot(adSlot);
+
+			callBidders(adSlot.getSlotName(), () => {
+				GptProvider.refreshSlot(adSlot);
+			});
+
 			window.googletag.pubads().removeEventListener('slotVisibilityChanged', refresh);
 		}
 	}
 	window.googletag.pubads().addEventListener('slotVisibilityChanged', refresh);
 }
 
-class SlotRefresher {
+export class SlotRefresher {
 	config: Config;
 	slotsInTheViewport: Array<string> = [];
 
 	log(...logValues) {
 		logger(logGroup, ...logValues);
 	}
 
-	refreshSlot(adSlot: AdSlot) {
-		if (!this.config.slots.includes(adSlot.getSlotName())) return;
+	isSlotRefreshable(slotName: string) {
+		return this.config.slots.includes(slotName);
+	}
+
+	/**
+	 * Removes adSlot sizes taller than the first rendered adSlot with the same slotName.
+	 * @param {AdSlot} adSlot - The ad slot to adjust.
+	 */
+	removeHigherSlotSizes(adSlot: AdSlot) {
+		const slotName = adSlot.getSlotName();
+		const slotHeight = adSlot.getElement().clientHeight;
+
+		const slotRefresherConfig = context.get('slotConfig.slotRefresher.sizes');
+		if (!slotRefresherConfig) return;
+
+		const slotHeightLimit = slotRefresherConfig[slotName][1];
+		if (!slotHeightLimit) return;
```

Comment:
[fun fact[
I know, its not elegant, but.... 
```suggestion
		if (!(slotHeightLimit = slotRefresherConfig[slotName][1])) { console.log('a'); }
```
did you thought about it? ;)